firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -422,65 +493,19 @@ private Query whereHelper(@NonNull FieldPath fieldPath, Operator op, Object valu
                 + op.toString()
                 + "' queries on FieldPath.documentId().");
       }
-      if (value instanceof String) {
-        String documentKey = (String) value;
-        if (documentKey.isEmpty()) {
-          throw new IllegalArgumentException(
-              "Invalid query. When querying with FieldPath.documentId() you must provide a valid "
-                  + "document ID, but it was an empty string.");
-        }
-        if (!query.isCollectionGroupQuery() && documentKey.contains("/")) {
-          throw new IllegalArgumentException(
-              "Invalid query. When querying a collection by FieldPath.documentId() you must "
-                  + "provide a plain document ID, but '"
-                  + documentKey
-                  + "' contains a '/' character.");
-        }
-        ResourcePath path = query.getPath().append(ResourcePath.fromString(documentKey));
-        if (!DocumentKey.isDocumentKey(path)) {
-          throw new IllegalArgumentException(
-              "Invalid query. When querying a collection group by FieldPath.documentId(), the "
-                  + "value provided must result in a valid document path, but '"
-                  + path
-                  + "' is not because it has an odd number of segments ("
-                  + path.length()
-                  + ").");
+      if (op == Operator.IN) {
+        validateDisjunctiveOperatorValueArray(value, op);
+        List referenceList = new ArrayList();
```

Comment:
Prefer generic versions of `List<>` for type-safety, so this would be `List<ReferenceValue>`.

Reply:
`ArrayValue.fromList()` requires `List<FieldValue>`, so I'm using that instead.

Reply:
FWIW, if we wanted to make this work, we could change `ArrayValue` to wrap a `List<? extends FieldValue>` instead of `List<FieldValue>` (read https://dzone.com/articles/covariance-and-contravariance for context).  Probably fine to leave as-is though.